src/packet/gre.rs.in

```diff
@@ -0,0 +1,125 @@
+// Copyright (c) 2016 Robert Collins <robertc@robertcollins.net>
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+#[cfg(test)]
+use packet::Packet;
+
+use packet::PrimitiveValues;
+
+use pnet_macros_support::types::*;
+
+/// GRE Packet.
+///
+/// Note that routing information from RFC 1701 is not implemented, packets
+/// with routing_present true will have incorrect payload lengths.
```

Comment:
How hard is it to implement this? I'd rather not have things being handled incorrectly.